src/components/pill/pill.pw.tsx

```diff
@@ -0,0 +1,427 @@
+  ] as const).forEach(([color, output]) => {
+    test(`should render borderColor prop set as ${output}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <PillComponent pillRole="status" borderColor={color}>
+          Pill
+        </PillComponent>
+      );
+
+      await checkCSSOutline(
+        pillPreview(page),
+        "2px",
+        "border",
+        "solid",
+        output
+      );
+    });
+  });
+
+  ["20px", "100px"].forEach((maxWidth) => {
+    test(`should render maxWidth prop set to ${maxWidth}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<Pill maxWidth={maxWidth}>Pill with a long label</Pill>);
+
+      const elementLocator = pillPreview(page);
+      await expect(elementLocator).toHaveCSS("max-width", maxWidth);
+    });
+  });
+
+  ([
+    [small, "16px", "12px", "0px 8px"],
+    [medium, "20px", "14px", "0px 8px"],
+    [large, "24px", "14px", "0px 8px"],
+    [extraLarge, "28px", "16px", "0px 12px"],
+  ] as const).forEach(([size, height, fontSize, padding]) => {
+    test(`should render size prop set as ${size}`, async ({ mount, page }) => {
+      await mount(<Pill size={size}>Pill</Pill>);
+
+      const elementLocator = pillPreview(page);
+
+      await expect(elementLocator).toHaveCSS("min-height", height);
+
+      await expect(elementLocator).toHaveCSS("line-height", height);
+
+      await expect(elementLocator).toHaveCSS("font-size", fontSize);
+
+      await expect(elementLocator).toHaveCSS("padding", padding);
+    });
+  });
+
+  ([
+    [true, "break-spaces"],
+    [false, "nowrap"],
+  ] as const).forEach(([booleanState, cssValue]) => {
+    test(`should render wrapText prop set as ${booleanState}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <Pill maxWidth="44px" wrapText={booleanState}>
+          Wrapped pill
+        </Pill>
+      );
+
+      const elementLocator = pillPreview(page);
+      await expect(elementLocator).toHaveText("Wrapped pill");
+      await expect(elementLocator).toHaveCSS("white-space", cssValue);
+    });
+  });
+});
+
+test.describe("should check focus outlines and border radius", () => {
+  (["S", "M", "L", "XL"] as PillProps["size"][]).forEach((size) => {
+    test(`should have the expected styling when size is ${size} and focusRedesignOptOut is false`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <PillComponent size={size} onDelete={() => {}}>
+          Pill
+        </PillComponent>
+      );
+
+      const elementLocator = pillCloseIcon(page);
+      await elementLocator.focus();
+      await expect(elementLocator).toHaveCSS(
+        "box-shadow",
+        "rgb(255, 188, 25) 0px 0px 0px 3px, rgba(0, 0, 0, 0.9) 0px 0px 0px 6px"
+      );
+      await expect(elementLocator).toHaveCSS(
+        "outline",
+        "rgba(0, 0, 0, 0) solid 3px"
+      );
+    });
+  });
+
+  (["S", "M", "L", "XL"] as PillProps["size"][]).forEach((size) => {
```

Comment:
As above, can you just use the constants `small`, `medium`, `large` and `extraLarge` instead of stating the values?